controllers/front/DiscountController.php

```diff
@@ -174,4 +174,35 @@ protected function accumulateCartRuleValue($voucher)
 
         return $cartRuleValue;
     }
+
+    /**
+     * @param array $voucher
+     *
+     * @return array
+     */
+    protected function buildCartRuleFromVoucher(array $voucher): array
+    {
+        $voucher['voucher_date'] = Tools::displayDate($voucher['date_to'], null, false);
+
+        if ($voucher['minimum_amount'] === 0) {
```

Comment:
Are you sure it's always an integer and not a string?

Reply:
I have no idea 🙅‍♂it's legacy. It should be an integer ... "should" 🤦‍♂

Reply:
```suggestion
        if ((int) $voucher['minimum_amount'] === 0) {
```
To be sure maybe?

Reply:
Deal 🤝